nengo_loihi/block.py

```diff
@@ -683,9 +685,11 @@ def _set_weights_indices(
         self,
         weights,
         indices=None,
-        weight_dtype=np.float32,
+        weight_dtype=None,
         compression=0,
     ):
+        # must be float32, because of how we discretize in place to int32
```

Comment:
Yeah, that comment might be a bit misleading. What I meant is of the float dtypes, only `float32` works. Essentially, it can only be `float32` if it's not yet discretized, and only `int32` if it's discretized. It looks like the only place we actually use the `weight_dtype` argument is in `split_blocks.py`, which currently is always called before we discretize, so I don't think we'd ever have `weight_dtype` other than `float32`. But I did want to write `split_blocks.py` so that it could theoretically work on a discretized model as well, in which case we'd be passing `weight_dtype=np.int32`.

Reply:
It might be good to add an explicit assertion that `weight_dtype` is either `float32` or `int32` (might want to pass it through `np.dtype` first, though, so that we can still handle e.g. strings properly).